### ISSUE TITLE
Discrepancies between Paper and Release Configuration

### ISSUE THREAD

**Issue by wengzejia1** [saidwivedi/TokenHMR#17]
Thanks for your amazing work!

I have a small concern regarding the hyperparameters. In Section 5.1 of your paper, it mentions that you used ViT-H/16 from ViTPose, trained for 100K steps, with a batch size of 256 and a learning rate of 1e-5.

However, when I check the configuration file [tokenhmr_release.yaml
](https://github.com/saidwivedi/TokenHMR/blob/main/tokenhmr/lib/configs_hydra/experiment/tokenhmr_release.yaml), I noticed the following differences:

- The total training steps are set to 350K instead of 100K.
- The ViT-H/16 backbone seems to be pre-trained with weights from HMR2.0, not from ViTPose.
- The batch size and learning rate are also different from those stated in the paper.

This leads to a significant difference in training time. The paper mentions that the model can be trained in about one day, while the configuration in the GitHub repo suggests that it might take around four days. 

Could you please clarify which set of hyperparameters should be used to reproduce the results from the paper? Can I follow the original paper’s hyperparameters for shorter training time and reproduce the paper results?

Thanks again for your hard work, and I look forward to your response!

**Comment by saidwivedi** [comment]
Thank you for your interest in our work :)

Yes, you are reproduce the paper results with the hyper-parameters mentioned in the paper - ViTPose backbone with higher learning rate (1e-5) and training for 100k iterations with effective batch size of 256. However, upon extensive testing of different settings before the code release we realized that 

- Model trained with HMR2.0 backbone with lower learning rate (5e-7) for 200k iterations not only gives similar 3D accuracy as one mentioned in the paper but also works better for diverse in-the-wild poses (see TokenHMR-ITW of Model zoo in README)
- Model with same setting as above but trained for 350k iterations gives slightly reduced 3D accuracy but with better image alignment (see TokenHMR-Demo of Model zoo in README).

One can effectively play with slightly higher learning rate (maybe 1e-6) to get similar performance but for reduced training iterations. And also depending on your need, you can do early stopping. 

Hope it clarifies :)

**Comment by saidwivedi** [comment]
It is the same model as TokenHMR-Demo which is trained for more iterations. You compromise slightly on 3D accuracy but get better 2D alignment.

**Comment by saidwivedi** [comment]
Since I do not have access to old environment, I trained a TokenHMR model with the Github released code with ViTPose pretrained model for 100K iterations. You can find the configuration [here](https://github.com/saidwivedi/TokenHMR/blob/main/tokenhmr/lib/configs_hydra/experiment/tokenhmr_paper.yaml). After 100K iterations, the results on 3DPW are-
`mode_re: 44.8 mm`
`mode_mpjpe: 70.9 mm`
`mode_pve: 84.3 mm`
I have also uploaded the checkpoint, please check the README.

If you change the number of GPUs or batch size, adjust the learning rate accordingly.

**Comment by wengzejia1** [comment]
Thanks a lot.
